yazi-core/src/position.rs

```diff
@@ -1,22 +1,47 @@
 use ratatui::prelude::Rect;
 
+#[derive(Debug, Clone, Copy)]
+pub struct RectShim {
+	pub x_offset: i16,
+	pub y_offset: i16,
+	pub width:    u16,
+	pub height:   u16,
+}
+
+impl Default for RectShim {
+	fn default() -> Self { Self { x_offset: 0, y_offset: 2, width: 50, height: 3 } }
+}
+
+#[derive(Debug, Clone)]
 pub enum Position {
-	Top(Rect),
-	Sticky(Rect, Rect),
-	Hovered(Rect),
+	TopLeft(RectShim),
```

Comment:
I changed this because `offset` could be a negative number. The parameters of `ratatui::prelude::Rect`  are positive coordinates.

Reply:
Maybe we can use a struct like`Offset { x, y, width, height }` with the better name of `Offset`